src/game_entities/shop.py

```diff
@@ -50,16 +51,18 @@ def __init__(
         name: str,
         position: Position,
         sprite_link: str,
+        shop_balance: int,
         stock: list[dict[str, any]],
         interaction: Optional[dict[str, any]] = None,
         sprite: Optional[pygame.Surface] = None,
     ) -> None:
         super().__init__(name, position, sprite_link, interaction, sprite)
+        self.shop_balance = 500
         self.current_visitor: Optional[Character] = None
         self.stock: list[dict[str, any]] = stock
         self.interaction: dict[str, any] = interaction
         self.menu: InfoBox = menu_creator_manager.create_shop_menu(
-            Shop.interaction_callback, self.stock, 0
+            Shop.interaction_callback, self.stock, 0, shop_balance
```

Comment:
Better relying on the attribute value already at this point, so `self.shop_balance`.